walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,179 @@
+        
+        ArrayList<Event> eventsCopy = new ArrayList<Event>();
+
+        eventsCopy = loadEvents(events);
+        
+        eventsCopy.sort(Comparator.comparing(Event::getWhen, TimeRange.ORDER_BY_START));
+
+
+        TimeRange tempT = new TimeRange(0, 24 * 60);
+
+    /// if our duration is more than a day 
+        if(request.getDuration() > tempT.duration() ){
+            return availableTimes; // return empty array 
+        }
+        else if(request.getAttendees().size() == 0){
+                 availableTimes.add(tempT);
+                 return availableTimes;
+        }
+        
+        int tempConflict = 0;
+        int previous_endTime = 0; 
+        boolean conflict = false;
+  
+        ArrayList<String> currentAttendees = new ArrayList<String>();
+        ArrayList<TimeRange> availableGaps = new ArrayList<TimeRange>();
+        availableGaps = getGaps(eventsCopy);
+        
+
+
+
+
+        for(Event eV : eventsCopy){
+
+
+            for(String attendee: eV.getAttendees()){
+                if(request.getAttendees().contains(attendee)){
+
+
+                        
+
+                        if(currentAttendees.contains(attendee)){
+                            conflict = true;
+                            continue;
+                            
+                        }
+
+                            
+                        
+
+                        if(previous_endTime < eV.getWhen().start()){
+                            
+                                TimeRange tempTr = new TimeRange(0, 24 * 60);
+                                tempTr = tempTr.fromStartEnd(previous_endTime,eV.getWhen().start() - 1,true);
+                                availableTimes.add(tempTr);
+                        }
+
+
+
+                        if(previous_endTime <  eV.getWhen().end()){
+                            previous_endTime = eV.getWhen().end();
+                        }
+                        
+                        
+
+                        if(overlaps(availableTimes,getTimeBefore(eV))){
+                       
+                        availableTimes.add(getTimeBefore(eV));
+                        }
+
+
+
+
+                        tempConflict = eV.getWhen().end();
+                        
+                        currentAttendees.add(attendee);
+
+                }
+                
+            }
+        }
+
+      
+        
+        availableTimes.add(getTimeAfter(previous_endTime));
+    
+ 
+    return availableTimes;
+    // return results;
+    }
+
+
+    public static ArrayList<Event> loadEvents(Collection<Event> events){
+         ArrayList<Event> result = new ArrayList<Event>();
+
+        for(Event ev : events){
+            result.add(ev);
+        }
+        return result;
+    }
+
+
+   public static TimeRange getTimeAfter(int start){
+       
+        TimeRange tempT = new TimeRange(start, 24 * 60);
+       tempT = tempT.fromStartEnd(start,(24 * 60) - 1,true);
```

Comment:
You can make this a non-inclusive time range (instead of subtracting 1 from it):
```java
return TimeRange.fromStartEnd(start, TimeRange.END_OF_DAY, true);
```